support/Environments/caltech_hpc_gcc.sh

```diff
@@ -0,0 +1,312 @@
+#%Module1.0
+prepend-path CPATH "$BOOST_HOME/include"
+setenv BOOST_ROOT "$BOOST_HOME"
+EOF
+    fi
+    cd $dep_dir
+
+    if [ -f $dep_dir/modules/spectre_gsl ]; then
+        echo "GSL is already set up"
+    else
+        # This hard-coded gsl path is not easily maintainable, so if and when
+        # Caltech HPC assigns an accessible environment variable for the loaded
+        # GSL module, we should use that in favor of the hard-coded path
+        cat >$dep_dir/modules/spectre_gsl <<EOF
+#%Module1.0
+setenv GSL_HOME "/software/gsl/2.4"
+setenv GSL_ROOT "/software/gsl/2.4"
+EOF
+    fi
+    cd $dep_dir
+
+    if [ -f $dep_dir/jemalloc/lib/libjemalloc.a ]; then
+        echo "jemalloc is already set up"
+    else
+        echo "Installing jemalloc..."
+        wget https://github.com/jemalloc/jemalloc/releases/download/5.2.1/jemalloc-5.2.1.tar.bz2
+        tar -xjf jemalloc-5.2.1.tar.bz2
+        mv jemalloc-5.2.1 jemalloc
+        cd $dep_dir/jemalloc
+        ./autogen.sh
+        make
+        cd $dep_dir
+        rm jemalloc-5.2.1.tar.bz2
+        echo "Installed jemalloc into $dep_dir/jemalloc"
+        cat >$dep_dir/modules/jemalloc <<EOF
+#%Module1.0
+prepend-path LIBRARY_PATH "$dep_dir/jemalloc/lib"
+prepend-path LD_LIBRARY_PATH "$dep_dir/jemalloc/lib"
+prepend-path CPATH "$dep_dir/jemalloc/include"
+prepend-path PATH "$dep_dir/jemalloc/bin"
+prepend-path LD_RUN_PATH "$dep_dir/jemalloc/lib"
+setenv JEMALLOC_HOME $dep_dir/jemalloc/
+prepend-path CMAKE_PREFIX_PATH "$dep_dir/jemalloc/"
+EOF
+    fi
+
+    cd $start_dir
+
+    spectre_unload_sys_modules
+
+    printf "\n\nIMPORTANT!!!\nIn order to be able to use these modules you\n"
+    echo "must run:"
+    echo "  module use $dep_dir/modules"
+    echo "You will need to do this every time you compile SpECTRE, so you may"
+    echo "want to add it to your ~/.bashrc."
+}
+
+spectre_unload_modules() {
+    module unload charm
+    module unload yaml-cpp
+    module unload spectre_boost
+    module unload spectre_gsl
+    module unload jemalloc
+    module unload libxsmm
+    module unload libsharp
+    module unload catch
+    module unload brigand
+    module unload blaze
+
+    spectre_unload_sys_modules
+}
+
+spectre_load_modules() {
+    spectre_load_sys_modules
+
+    module load blaze
+    module load brigand
+    module load catch
+    module load libsharp
+    module load libxsmm
+    module load jemalloc
+    module load spectre_boost
+    module load spectre_gsl
+    module load yaml-cpp
+    module load charm
+}
+
+spectre_run_cmake() {
+    if [ -z ${SPECTRE_HOME} ]; then
+        echo "You must set SPECTRE_HOME to the cloned SpECTRE directory"
+        return 1
+    fi
+    spectre_load_modules
+    cmake -D CHARM_ROOT=$CHARM_ROOT \
+          -D CMAKE_BUILD_TYPE=Release \
+          -D CMAKE_Fortran_COMPILER=gfortran \
+          -D MEMORY_ALLOCATOR=JEMALLOC \
+          -D BUILD_PYTHON_BINDINGS=off \
+          -D PYTHON_EXECUTABLE=`which python3` \
```

Comment:
This should be `Python_EXECUTABLE`. Not sure if cmake has some failsafes in place to make this capitalization work as well.